src/reconcile/rec_visibility.c

```diff
@@ -637,30 +637,68 @@ __rec_calc_upd_memsize(WT_UPDATE *onpage_upd, WT_UPDATE *tombstone, size_t upd_m
 
 /*
  * __rec_upd_select --
- *     Select the update to write to disk image.
+ *     Select the update to write to disk image. @param write_prepare True if we should write the
+ *     update as a prepared update (prepare timestamp is stable but durable timestamp is not), false
+ *     to write as committed.
  */
 static int
 __rec_upd_select(WT_SESSION_IMPL *session, WTI_RECONCILE *r, WT_UPDATE *first_upd,
   WTI_UPDATE_SELECT *upd_select, WT_UPDATE **first_txn_updp, bool *has_newer_updatesp,
-  size_t *upd_memsizep)
+  bool *write_prepare, size_t *upd_memsizep)
 {
     WT_CONNECTION_IMPL *conn;
-    WT_UPDATE *upd;
+    WT_UPDATE *upd, *prepare_rollback_tombstone;
     wt_timestamp_t max_ts;
     uint64_t max_txn, session_txnid, txnid;
     uint8_t prepare_state;
     bool is_hs_page, seen_prepare;
 
     conn = S2C(session);
+    prepare_rollback_tombstone = NULL;
     max_ts = WT_TS_NONE;
     max_txn = WT_TXN_NONE;
     is_hs_page = F_ISSET(session->dhandle, WT_DHANDLE_HS);
     session_txnid = __wt_atomic_loadv64(&WT_SESSION_TXN_SHARED(session)->id);
     seen_prepare = false;
+    *write_prepare = false;
 
     for (upd = first_upd; upd != NULL; upd = upd->next) {
-        if ((txnid = upd->txnid) == WT_TXN_ABORTED)
-            continue;
+        *write_prepare = false;
```

Comment:
Can we add a comment here to describe why we need to reset it here?

Reply:
done